src/coreclr/jit/emitfmtsarm64_sve.h

```diff
@@ -0,0 +1,514 @@
+IF_DEF(SVE_AL_3A,   IS_NONE, NONE) // SVE_AL_3A  ........xx...... ...gggnnnnnddddd  -- sve_int_reduce_1q
+IF_DEF(SVE_AM_2A,   IS_NONE, NONE) // SVE_AM_2A  ........xx...... ...gggxxiiiddddd  -- sve_int_bin_pred_shift_0
+IF_DEF(SVE_AN_3A,   IS_NONE, NONE) // SVE_AN_3A  ........xx...... ...gggmmmmmddddd  -- sve_int_bin_pred_shift_1
+IF_DEF(SVE_AO_3A,   IS_NONE, NONE) // SVE_AO_3A  ........xx...... ...gggmmmmmddddd  -- sve_int_bin_pred_shift_2
+IF_DEF(SVE_AP_3A,   IS_NONE, NONE) // SVE_AP_3A  ........xx...... ...gggnnnnnddddd  -- sve_int_un_pred_arit_1
+IF_DEF(SVE_AQ_3A,   IS_NONE, NONE) // SVE_AQ_3A  ........xx...... ...gggnnnnnddddd  -- sve_int_un_pred_arit_0
+IF_DEF(SVE_AR_4A,   IS_NONE, NONE) // SVE_AR_4A  ........xx.mmmmm ...gggnnnnnddddd  -- sve_int_mlas_vvv_pred
+IF_DEF(SVE_AS_4A,   IS_NONE, NONE) // SVE_AS_4A  ........xx.mmmmm ...gggaaaaaddddd  -- sve_int_mladdsub_vvv_pred
+IF_DEF(SVE_AT_3A,   IS_NONE, NONE) // SVE_AT_3A  ........xx.mmmmm ......nnnnnddddd  -- sve_int_bin_cons_arit_0
+IF_DEF(SVE_AT_3B,   IS_NONE, NONE) // SVE_AT_3B  ...........mmmmm ......nnnnnddddd  -- sve_int_bin_cons_arit_0
+IF_DEF(SVE_AU_3A,   IS_NONE, NONE) // SVE_AU_3A  ...........mmmmm ......nnnnnddddd  -- sve_int_bin_cons_log
+IF_DEF(SVE_AV_3A,   IS_NONE, NONE) // SVE_AV_3A  ...........mmmmm ......kkkkkddddd  -- sve_int_tern_log
+IF_DEF(SVE_AW_2A,   IS_NONE, NONE) // SVE_AW_2A  ........xx.xxiii ......mmmmmddddd  -- sve_int_rotate_imm
+IF_DEF(SVE_AX_1A,   IS_NONE, NONE) // SVE_AX_1A  ........xx.iiiii ......iiiiiddddd  -- sve_int_index_ii
+IF_DEF(SVE_AY_2A,   IS_NONE, NONE) // SVE_AY_2A  ........xx.mmmmm ......iiiiiddddd  -- sve_int_index_ir
+IF_DEF(SVE_AZ_2A,   IS_NONE, NONE) // SVE_AZ_2A  ........xx.iiiii ......nnnnnddddd  -- sve_int_index_ri
+IF_DEF(SVE_BA_3A,   IS_NONE, NONE) // SVE_BA_3A  ........xx.mmmmm ......nnnnnddddd  -- sve_int_index_rr
+IF_DEF(SVE_BB_2A,   IS_NONE, NONE) // SVE_BB_2A  ...........nnnnn .....iiiiiiddddd  -- sve_int_arith_vl
+IF_DEF(SVE_BC_1A,   IS_NONE, NONE) // SVE_BC_1A  ................ .....iiiiiiddddd  -- sve_int_read_vl_a
+IF_DEF(SVE_BD_3A,   IS_NONE, NONE) // SVE_BD_3A  ........xx.mmmmm ......nnnnnddddd  -- sve_int_mul_b
+IF_DEF(SVE_BD_3B,   IS_NONE, NONE) // SVE_BD_3B  ...........mmmmm ......nnnnnddddd  -- sve_int_mul_b
+IF_DEF(SVE_BE_3A,   IS_NONE, NONE) // SVE_BE_3A  ........xx.mmmmm ......nnnnnddddd  -- sve_int_sqdmulh
+IF_DEF(SVE_BF_2A,   IS_NONE, NONE) // SVE_BF_2A  ........xx.xxiii ......nnnnnddddd  -- sve_int_bin_cons_shift_b
+IF_DEF(SVE_BG_3A,   IS_NONE, NONE) // SVE_BG_3A  ........xx.mmmmm ......nnnnnddddd  -- sve_int_bin_cons_shift_a
+IF_DEF(SVE_BH_3A,   IS_NONE, NONE) // SVE_BH_3A  .........x.mmmmm ....hhnnnnnddddd  -- sve_int_bin_cons_misc_0_a
+IF_DEF(SVE_BH_3B,   IS_NONE, NONE) // SVE_BH_3B  ...........mmmmm ....hhnnnnnddddd  -- sve_int_bin_cons_misc_0_a
+IF_DEF(SVE_BH_3B_A, IS_NONE, NONE) // SVE_BH_3B_A  ...........mmmmm ....hhnnnnnddddd  -- 
```

Comment:
Why do some forms, like this one, have an extra discriminator (here, `_A`) at the end? This looks identical in pattern to the preceding one.

Reply:
Yes, they share the same "instruction format" as in ` ...........mmmmm ...gggnnnnnttttt`, but the actual opcode i.e. 0 and 1 are different for them. You can see them in `instrsarm64_sve.h`:

```
    // ST1D    {<Zt>.D }, <Pg>, [<Xn|SP>, <Xm>, LSL #3]                                  SVE_JD_4C           11100101111mmmmm 010gggnnnnnttttt     E5E0 4000   
    // ST1D    {<Zt>.Q }, <Pg>, [<Xn|SP>, <Xm>, LSL #3]                                  SVE_JD_4C_A         11100101110mmmmm 010gggnnnnnttttt     E5C0 4000   

```

It is possible to combine them to be something like ` ..........Xmmmmm ...gggnnnnnttttt`, where `X` in this pattern is the differing bit, but the tool won't know if logically they can be combined or not, so I kept it simple and gave it a new name. The reason for appending `_A` or `_B` at the end is so we can having unique format names with the instruction. We cannot have duplicate format names for any instruction because while searching, we won't know which of the formats we wanted to get hex code for.

Reply:
I presume, if the tool can't automatically merge them, we would want a developer coming in afterwards to manually merge them and add the `X` (for example) bit? At least in this case?

Reply:
That's correct. So for an instruction, if there are 2 formats `SVE_AA` and `SVE_AB` that are worth merging into `SVE_AA`, we do not have to delete the entry of `SVE_AB` because it can be still used for other instructions. Once there are no usage of `SVE_AB`, we can delete its entry from format table.